### ISSUE TITLE
improve fbopt performance of fbopt on diva by tuning KI GAIN

### ISSUE THREAD

**Issue by smilesun** [marrlab/DomainLab#596]
The performance here is ok compared to baslines,  https://github.com/marrlab/DomainLab/issues/578

This task is to tune the vital parameters of fbopt and change yaml file based on https://github.com/marrlab/DomainLab/blob/fbopt/examples/benchmark/pacs_diva_fbopt_alone_es1.yaml
and https://github.com/marrlab/DomainLab/blob/fbopt/examples/benchmark/pacs_diva_fbopt_alone_fixed.yaml (where KI-GAIN is fixed to current default value here https://github.com/marrlab/DomainLab/blob/dfd64cc383b18338ca26da942f796bee014d026e/domainlab/algos/trainers/args_fbopt.py#L11

**Comment by DanScarc** [comment]
Maybe 

`max: 0.1`

And 

`num: 4`

Since we have 2 values for every 10-fold increase? Or shall we increase this ratio to three values per 10-fold (i.e. `num: 6` in total)? What about runtime with this expansion?

**Comment by smilesun** [comment]
what you could do is to try a setting you like and post the hyperparameters.csv file here (use run_benchmark_slurm.sh xx.yaml and control c to kill the process after 1 second so that only hyperparameters.csv is generated in the zoutput folder). I think we could fix the other parameters like mu_init=1e-6(we support scientific notation now!) , mu_clip=1e3, we just tune k_i_gain and exp_shoulder_cap,  then we look at the tensorboard plot (post it here) so that we know the effect of k_i_gain and exp_shoulder_cap,  I am working on https://github.com/marrlab/DomainLab/issues/520 to make this automatic

**Comment by smilesun** [comment]
do **not** use run_benchmark_slurm but run_benchmark_standalone to generate the hyperparameters.csv file

**Comment by DanScarc** [comment]
Current hyperparameters: [hyperparameters.csv](https://github.com/marrlab/DomainLab/files/13177164/hyperparameters.csv)

**Comment by DanScarc** [comment]
![stochastic_variation](https://github.com/marrlab/DomainLab/assets/84813873/545aab14-7ef6-4ca3-af04-5cc9fa288f17)
[results.csv](https://github.com/marrlab/DomainLab/files/13186181/results.csv)

**Comment by smilesun** [comment]
could you run examples/benchmark/pacs_diva_fbopt_alone_es1_autoki.yaml

**Comment by DanScarc** [comment]
Is the seed important here?

**Comment by smilesun** [comment]
yes, please put any starting seed, e.g. 0

**Comment by smilesun** [comment]
`bash submit_slurm_via_cpu_10days.sh examples/benchmark/pacs_diva_fbopt_alone_es1_autoki_output_ma_9.yaml 0`

**Comment by DanScarc** [comment]
how do I check now whether slurm throws any errors?

**Comment by smilesun** [comment]
no matter if one use submission node or a separate cpu node to submit the jobs,  it is always the same to check if there is error:

use **any** of the 3 options
1.  use `sh_list_errors.sh zoutput/slurm_logs` 
2.  use ``squeue -u yourname` to see if jobs are still running (less reliable, since maybe jobs with bugs are being run )
3. check zoutput/benhcmarks/nameofbenchmark/logs

**Comment by DanScarc** [comment]
Excellent, then sanity check passed ✅

**Comment by DanScarc** [comment]
[rst.csv](https://github.com/marrlab/DomainLab/files/13202148/rst.csv)

**Comment by DanScarc** [comment]
Yes, looks good. 

> so our algorithm is multiplier free now

Yes :)

> robust to different hyper-parameters

Looks like that but we'd have to assess robustness more quantitatively. To a first inspection, there are no drastic drops in accuracy across the different combinations and mean looks > 70% across all seeds and settings 👍🏼

**Comment by smilesun** [comment]
I switched the issue to #606, i guess it is some ireproducible problem due to the simultainous occupation of resources, not something that we coudl resolve
